src/System.ComponentModel.TypeConverter/src/System/ComponentModel/VersionConverter.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    ///    <para>Provides a type converter to convert Version objects to and
+    ///       from various other representations.</para>
+    /// </summary>
+    public class VersionConverter : TypeConverter
+    {
+        /// <summary>
+        ///    <para>Gets a value indicating whether this converter can convert an object in the
+        ///       given source type to a Version.</para>
+        /// </summary>
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            if (sourceType == null)
+                throw new ArgumentNullException(nameof(sourceType));
+
+            return sourceType == typeof(string) || sourceType == typeof(Version);
+        }
+
+        /// <summary>
+        ///    <para>Gets a value indicating whether this converter can
+        ///       convert an object to the given destination type using the context.</para>
+        /// </summary>
+        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+        {
+            return destinationType == typeof(string) || destinationType == typeof(Version);
+        }
+
+        /// <summary>
+        ///    <para>Converts the given object to a Version.</para>
+        /// </summary>
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            if (value is string versionString)
+            {
+                // Let the Version constructor throw any informative exceptions
+                return new Version(versionString);
```

Comment:
It looks like other ConvertFrom methods trim the input string before trying to parse it.  Could you check what the behavior is there, and then potentially adapt this accordingly to match if it should trim?  Same question for IsValid

Reply:
The default behavior of Version.Parse is to ignore all whitespaces, so this is actually already taken care of. I added some test data in cfc4ea107f2807a6d84d41cf15a9c5ab8d1daa73 to ensure correct behavior.

Reply:
Ok, thanks.